src/utils/spawning.rs

```diff
@@ -382,6 +392,20 @@ mod systemd {
 
         let _ = write!(scope_name, "-{child_pid}.scope");
 
+        let mut slice_name = format!("app-niri-");
+
+        // Escape for systemd similarly to libgnome-desktop, which says it had adapted this from
+        // systemd source.
+        for &c in name.as_bytes() {
+            if c.is_ascii_alphanumeric() || matches!(c, b':' | b'_' | b'.') {
+                slice_name.push(char::from(c));
+            } else {
+                let _ = write!(slice_name, "\\x{c:02x}");
+            }
+        }
+
+        let _ = write!(slice_name, ".slice");
```

Comment:
Why a slice and not a scope btw? All apps are in `app.slice`, and I only have the few pre-created slices on my system.
```
└─ systemctl --user | grep .slice
  -.slice
  app.slice
  background.slice
  session.slice
  user.slice
```

Reply:
It should be both, the scope is named like `app-niri-<app_id>-{pid}.scope` and the slice is named `app-niri-<app_id>.slice`. This allows to override limits per `app_id`